src/lib.rs

```diff
@@ -3,31 +3,28 @@ use std::collections::HashMap;
 use micro_rdk::DoCommand;
 use micro_rdk::common::config::ConfigType;
 use micro_rdk::common::status::{Status, StatusError};
-use micro_rdk::common::registry::{ComponentRegistry, RegistryError, Dependency};
-use micro_rdk::common::i2c::I2cHandleType
+use micro_rdk::common::registry::{ComponentRegistry, RegistryError, Dependency, get_board_from_dependencies};
 
 // correct crates?
-use micro_rdk::board::Board
-
-use micro_rdk::common::sensor::{Sensor, SensorType, Readings, SensorError};
+use micro_rdk::common::board::Board;
+use micro_rdk::common::sensor::{Sensor, SensorType, Readings, SensorError, GenericReadingsResult, SensorResult, TypedReadingsResult, SensorT};
+use micro_rdk::common::analog::AnalogError;
```

Comment:
What is analog error being used for?

Reply:
I thought it would be more accurate to the type of error rather than giving a generic E (which didn't compile)

Reply:
SensorError is more accurate I believe, AnalogError would be related to an analog reader off our board interface in rust.

Reply:
I chnaged it to SensorError in a more recent commit, going to close this PR for now.